internal/resources/iampolicy/scope.go

```diff
@@ -0,0 +1,138 @@
+/*
+Copyright © 2022 VMware, Inc. All Rights Reserved.
+SPDX-License-Identifier: MPL-2.0
+*/
+
+package iampolicy
+
+import (
+	"fmt"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+
+	clustermodel "github.com/vmware-tanzu/terraform-provider-tanzu-mission-control/internal/models/cluster"
+	clustergroupmodel "github.com/vmware-tanzu/terraform-provider-tanzu-mission-control/internal/models/clustergroup"
+	namespacemodel "github.com/vmware-tanzu/terraform-provider-tanzu-mission-control/internal/models/namespace"
+	organizationmodel "github.com/vmware-tanzu/terraform-provider-tanzu-mission-control/internal/models/organization"
+	workspacemodel "github.com/vmware-tanzu/terraform-provider-tanzu-mission-control/internal/models/workspace"
+)
+
+type scopeType int
+
+const (
+	unknown scopeType = iota
+	organization
+	clusterGroup
+	cluster
+	workspace
+	namespace
+)
+
+var (
+	stringToScope = map[string]scopeType{
+		orgKey:          organization,
+		clusterGroupKey: clusterGroup,
+		clusterKey:      cluster,
+		workspaceKey:    workspace,
+		namespaceKey:    namespace,
+	}
+)
+
+func (s scopeType) String() string {
+	return [...]string{
+		"unknown",
+		orgKey,
+		clusterGroupKey,
+		clusterKey,
+		workspaceKey,
+		namespaceKey,
+	}[s]
```

Comment:
Can we use an int enum instead?

Do we need these as strings for some specific use case?

Reply:
The way function has been defined, it changes the string scope entered by the user to scopeType.

Reply:
I don't see a requirement to use a string here when an enum has already been used.